homeassistant/components/dnsip/config_flow.py

```diff
@@ -0,0 +1,78 @@
+"""Adds config flow for dnsip integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_HOSTNAME,
+    CONF_IPV6,
+    CONF_RESOLVER,
+    CONF_RESOLVER_IPV6,
+    DEFAULT_HOSTNAME,
+    DEFAULT_IPV6,
+    DEFAULT_NAME,
+    DEFAULT_RESOLVER,
+    DEFAULT_RESOLVER_IPV6,
+    DOMAIN,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOSTNAME, default=DEFAULT_HOSTNAME): cv.string,
+        vol.Optional(CONF_RESOLVER, default=DEFAULT_RESOLVER): cv.string,
```

Comment:
Is the resolver needed for the config flow?

Reply:
So you mean not to give the option to the user at all and just use the default resolver?

Reply:
No, I didn't say that.

Configuration flows, should ask the bare minimum to set up an integration.
Other things can assume defaults and can be made configurable using options flows instead.